daemon/openrazer_daemon/hardware/keyboards.py

```diff
@@ -881,6 +881,31 @@ class RazerCynosaChroma(_RippleKeyboard):
     }
 
 
+class RazerCynosaChromaPro(_MacroKeyboard):
+    """
+    Class for the Razer Cynosa Chroma Pro
+    """
+    EVENT_FILE_REGEX = re.compile(r'.*Cynosa_Chroma_Pro(-if01)?-event-kbd')
```

Comment:
These are the files that exist:
```
lrwxrwxrwx 1 root root  10 Dec  1 06:35 usb-Razer_Razer_Cynosa_Chroma_Pro-event-kbd -> ../event18
lrwxrwxrwx 1 root root  10 Dec  1 06:35 usb-Razer_Razer_Cynosa_Chroma_Pro-if01-event-kbd -> ../event19
lrwxrwxrwx 1 root root  10 Dec  1 06:35 usb-Razer_Razer_Cynosa_Chroma_Pro-if02-event-mouse -> ../event20
lrwxrwxrwx 1 root root   9 Dec  1 06:35 usb-Razer_Razer_Cynosa_Chroma_Pro-if02-mouse -> ../mouse4
```
So it looks sane... It would be great if you could get ripple working though, I'm not familiar with any part of the ripple code so I can't help you much.

Reply:
Yeah, I'm not familiar with the ripple effect either. I thought it would have worked fine. I can try and take a look at it :/

Reply:
On top of what you verified with the file name sanity, I couldn't find anything strange in ripple_effect.py.

And the ripple effect is currently working on my BlackWidow Chroma V2, which seems about the same as the Cynosa Chroma Pro functionality-wise. Both have the same ripple effect methods listed, matrix enabled, same matrix dimensions, and extend _RippleKeyboard.

Ripple also works for me in both RazerGenie and Polychromatic.

Unless there is something strange with the Cynosa Chroma Pro hardware and/or @gabrielfnd can provide error messages I'm at a loss.